### PR TITLE
Feature/add voucher options to match orders

### PR DESCRIPTION
**Pull request by abbesBenayache**

### CODE REVIEW

File: src/cli/cmd/iexec-app.js [hunk 1]
```diff
@@ -243,7 +243,8 @@ show
       if (!isAddress && !userAddress)
         throw Error(`Missing option ${option.user()[0]} or wallet`);
 
-      if (!addressOrIndex) throw Error(info.missingAddressOrDeployed(objName, chain.id));
+      if (!addressOrIndex)
```
Comment (abbesBenayache) [inline src/cli/cmd/iexec-app.js:246]: this change is due to the formatter on save and can be ignored.

File: src/cli/cmd/iexec-order.js [hunk 2]
```diff
@@ -516,13 +516,15 @@ fill
...
-          ] : [],
+          ...(useDataset
+            ? [
+                (
+                  await datasetorderSchema()
+                    .label('datasetorder')
+                    .validate(datasetOrder)
+                ).tag,
+              ]
+            : []),
```
Comment (abbesBenayache) [inline src/cli/cmd/iexec-order.js:527]: this change is due to the formatter on save and can be ignored.

File: src/common/market/order.js [hunk 3]
```diff
@@ -841,12 +845,104 @@ const getMatchableVolume = async (
...
+      balance,
+      isAppEligibleToMatchOrdersSponsoring,
+      isDatasetEligibleToMatchOrdersSponsoring,
+      isWorkerpoolEligibleToMatchOrdersSponsoring,
+    );
+  }
+
+  return { total: totalCost, sponsored: sponsoredCost };
+};
+
```
Comment (abbesBenayache) [inline src/common/market/order.js:937]: This change is due to moving the previously implemented estimateMatchOrder function to be used in matchOrder. the removed part is visible at the bottom of the changes.

File: src/common/market/order.js [hunk 4]
```diff
@@ -1081,93 +1238,3 @@ export const createRequestorder = async (
...
-      matchableVolume,
-      balance,
-      isAppEligibleToMatchOrdersSponsoring,
-      isDatasetEligibleToMatchOrdersSponsoring,
-      isWorkerpoolEligibleToMatchOrdersSponsoring,
-    );
-  }
-
-  return { total: totalCost, sponsored: sponsoredCost };
-};
```
Comment (abbesBenayache) [inline src/common/market/order.js:1173]: the previously relocated section can be found further up, as indicated in the preceding comment.

**Review by PierreJeanjacquot — CHANGES_REQUESTED** [review]

**Comment by PierreJeanjacquot** [comment]
proposed refacto avoiding abi duplication https://github.com/iExecBlockchainComputing/iexec-sdk/pull/266

**Review by PierreJeanjacquot — APPROVED** [review]

**Pull request merged**